### PR TITLE
Add outlier analysis

### PR DESCRIPTION
**Pull request by caldwellst**
Some new graphs on outliers. Simply looking at the relationship between the size and numbers of units of analysis, really.

### CODE REVIEW

File: 04_testing/outlier_checks.R [hunk 1]
```diff
@@ -897,3 +898,365 @@ write_csv(
...
+      breaks = 0:16,
+      labels = log_labs,
+      include.lowest = TRUE
+    )
+  ) %>%
+  group_by(
+    log_pop_bins
+  ) %>%
+  summarize(
+    is_any_outlier = is_any_outlier / n(),
```
Comment (Rawa-Ganjani) [inline 04_testing/outlier_checks.R:932]: in my computer the dataset stays the same size with the same 16 percentages being repeated, added sum to is_any_outlier then it reduced to the 16 bins, it's probably more efficient if there are no duplication

File: 04_testing/outlier_checks.R [hunk 2]
```diff
@@ -897,3 +898,365 @@ write_csv(
...
+    adm0_pcode
+  ) %>%
+  summarize(
+    units = n(),
+    num_sectors = length(unique(sector)),
+    pct_outliers = sum(is_any_outlier) / units,
+    .groups = "drop"
+  )
+
+df_outliers_num %>%
```
Comment (Rawa-Ganjani) [inline 04_testing/outlier_checks.R:1119]: showing the countries as legend would probably explain the crazy figures from the outlier graphs.

File: 04_testing/outlier_checks.R [hunk 3]
```diff
@@ -897,3 +898,365 @@ write_csv(
...
+    "graphs",
+    "Option 1",
+    "datasets",
+    "2022_outliers_number_units_analysis.csv"
+  )
+)
+
+# also plotting against # of sectors
+
+df_outliers_num %>%
```
Comment (Rawa-Ganjani) [inline 04_testing/outlier_checks.R:1192]: same here, showing the countries as legend will explain the other outlier graphs.

Comment (Rawa-Ganjani) [reply 04_testing/outlier_checks.R:1192]: maybe adding a note, normally for outlier detection, the more cases available the more accurate outlier detection will be. I know that it but don't have any proof for it yet 😆 , if you think it's correct having that will also validates the crazy numbers for countries with only the top common sectors.

File: 04_testing/outlier_checks.R (hunk 1 above)
Comment (caldwellst) [reply 04_testing/outlier_checks.R:932]: Good catch!

File: 04_testing/outlier_checks.R (hunk 2 above)
Comment (caldwellst) [reply 04_testing/outlier_checks.R:1119]: Hmm, do we think it's only country level phenomena and there's no general trend? What do you think?

File: 04_testing/outlier_checks.R (hunk 3 above)
Comment (caldwellst) [reply 04_testing/outlier_checks.R:1192]: Hmm from the graph, it appears that there's very few outliers if we only have a small # of sectors. What do we think about that?

**Pull request merged**